### ISSUE TITLE
Add `sobel_edges` to keras.ops.image

### ISSUE THREAD

**Issue by timon0305** [keras-team/keras#22148]
Add `sobel_edges` operation to `keras.ops.image` for computing Sobel edge detection on images.

Sobel edge detection is a fundamental image processing operation used for edge detection, gradient computation, and preprocessing in computer vision tasks.

TensorFlow provides [`tf.image.sobel_edges`](https://www.tensorflow.org/api_docs/python/tf/image/sobel_edges). Adding this to Keras ops would bring feature parity while supporting all backends.
